fs.go

```diff
@@ -39,21 +39,14 @@ type Filesystem interface {
 	MkdirAll(filename string, perm os.FileMode) error
 	Join(elem ...string) string
 	Dir(path string) Filesystem
-	Base() string
-}
-
-// Symlinker is a Filesystem with support for creating symlinks.
-type Symlinker interface {
```

Comment:
Including the Symlink function in the base billy.Filesystem interface is a backwards incompatible change.
Adding a separate interface allows us to have both implementations that support it and implementations that don't without breaking anything.
It would be good if we can preserve backwards compatibility, and then in the future consolidate API changes in billy v2.